include/ck_tile/ref/naive_grouped_conv_bwd_data.hpp

```diff
@@ -0,0 +1,362 @@
+                            {
+                                ck_tile::long_index_t out_idx = n * out_strides[0] +
+                                                                wo * out_strides[1] +
+                                                                g * out_strides[2] + k;
+                                ck_tile::long_index_t wei_idx =
+                                    g * wei_strides[0] + k * wei_strides[1] + x * wei_strides[2] +
+                                    c * wei_strides[3];
+
+                                v_acc += type_convert<float>(p_out_grad[out_idx]) *
+                                         type_convert<float>(p_wei[wei_idx]);
+                            }
+                        }
+                    }
+                }
+                else if constexpr(NDimSpatial == 2)
+                {
+                    for(ck_tile::index_t y = 0; y < wei_spatial_lengths[0]; ++y)
+                    {
+                        ck_tile::long_index_t h_tmp =
+                            static_cast<ck_tile::long_index_t>(in_spatial_idx[0]) +
+                            static_cast<ck_tile::long_index_t>(in_left_pads[0]) -
+                            static_cast<ck_tile::long_index_t>(y * conv_dilations[0]);
+
+                        if(h_tmp % conv_strides[0] == 0)
+                        {
+                            ck_tile::long_index_t ho = h_tmp / conv_strides[0];
+
+                            if(ho >= 0 && ho < out_spatial_lengths[0])
+                            {
+                                for(ck_tile::index_t x = 0; x < wei_spatial_lengths[1]; ++x)
+                                {
+                                    ck_tile::long_index_t w_tmp =
+                                        static_cast<ck_tile::long_index_t>(in_spatial_idx[1]) +
+                                        static_cast<ck_tile::long_index_t>(in_left_pads[1]) -
+                                        static_cast<ck_tile::long_index_t>(x * conv_dilations[1]);
+
+                                    if(w_tmp % conv_strides[1] == 0)
+                                    {
+                                        ck_tile::long_index_t wo = w_tmp / conv_strides[1];
+
+                                        if(wo >= 0 && wo < out_spatial_lengths[1])
+                                        {
+                                            ck_tile::long_index_t out_idx =
+                                                n * out_strides[0] + ho * out_strides[1] +
+                                                wo * out_strides[2] + g * out_strides[3] + k;
+                                            ck_tile::long_index_t wei_idx =
+                                                g * wei_strides[0] + k * wei_strides[1] +
+                                                y * wei_strides[2] + x * wei_strides[3] +
+                                                c * wei_strides[4];
```

Comment:
Here and for 1d,2d,3d: could you move index calculation to some function? It could be common function for each NumDSpatial or some overloading